Modules/audioop.c

```diff
@@ -1364,10 +1368,18 @@ audioop_ratecv_impl(PyObject *module, Py_buffer *fragment, int width,
             goto exit;
         }
         for (chan = 0; chan < nchannels; chan++) {
-            if (!PyArg_ParseTuple(PyTuple_GetItem(samps, chan),
-                                  "ii:ratecv", &prev_i[chan],
-                                               &cur_i[chan]))
+            channel = PyTuple_GetItem(samps, chan);
+            if (!PyTuple_Check(channel)) {
+                PyErr_SetString(PyExc_TypeError,
+                                "ratecv(): channel must be a tuple");
```

Comment:
i am not sure about this error message, as I failed to find the documentation of the state argument.